### ISSUE TITLE
Dropdown panel keyboard navigation is not correct with custom option template

### ISSUE THREAD

**Issue by kalpeshhpatel** [ng-select/ng-select#1940]
**Describe the bug**

When the custom template is used for the dropdown item and the data is loaded asynchronously, the keyboard navigation with up/down arrow key is not correctly scrolling to the selected item.

**Reproducbile example**

https://stackblitz.com/edit/angular-rlbjcy?file=src/search-autocomplete-example.component.ts

**To Reproduce** 
Steps to reproduce the behavior (if example is not provided):
1. Go to 'https://stackblitz.com/edit/angular-rlbjcy?file=src/search-autocomplete-example.component.ts'
2. Reload the application preview
3. Start typing in the search panel to load users.
4. The users will be shown in the dropdown panel with custom template.
5. Use down arrow key to navigate.
6. Observe that top scroll position of selected item is not correct.

**Expected behavior**
With custom template too, the position of selected element with keyboard navigation should be correct.

**Screenshots**

![keyboard-navigation-issue](https://user-images.githubusercontent.com/5223493/127466894-7bea6193-100c-4618-b6c3-71de1d7752a4.gif)

![Screen Shot 2021-07-29 at 1 22 36 PM](https://user-images.githubusercontent.com/5223493/127453466-3a23bdd6-2ef7-43fa-9905-2d16dacfe6a6.png)

Possible Root Cause:

This piece of code: https://github.com/ng-select/ng-select/blob/master/src/ng-select/lib/ng-dropdown-panel.component.ts#L264 is executed only once. However when the data has received from the API asynchronously and the list is updated, the `panelHeight` needs to be calculated again.

That is not happening currently and during the calculation here: https://github.com/ng-select/ng-select/blob/master/src/ng-select/lib/ng-dropdown-panel.service.ts#L67 the older panelHeight is getting used which results in wrong scroll top position.
